pkg/blockbuilder/scheduler/scheduler.go

```diff
@@ -65,6 +77,37 @@ func (s *BlockBuilderScheduler) stopping(_ error) error {
 }
 
 func (s *BlockBuilderScheduler) running(ctx context.Context) error {
+	// The startup process for block-builder-scheduler entails learning the state of the world:
+	//  1. obtain an initial set of offset info from Kafka
+	//  2. listen to worker updates for a while to learn what the previous scheduler knew
+	// When both of those are complete, we transition from observation mode to normal operation.
```

Comment:
(nit) What if we move these pieces of work into the `starting` handler? This may better separate the logic before and after the scheduler becomes ready to do the work.

Reply:
I like the idea, hadn't considered it. The service needs to be "up" during this time to receive requests, but reading the BasicService code, that wouldn't interfere there. Done.